src/Services/PrepareServiceInterface.php

```diff
@@ -39,7 +40,7 @@ public function transferLazy(
         Wallet $from,
         Wallet $to,
         string $status,
-        float|int|string $amount,
+        CostDtoInterface|float|int|string $amount,
```

Comment:
Must accept a "price" group. Forms two checks, replenishment and write-off of funds.
You need to return the currency in which the payment was made at the "previous rate" (maybe even worth setting it up). And you need to write off the product in its currency.